### PR TITLE
#40 #43 #46 Listing members of room

### PR DESCRIPTION
**Pull request by michaelKurowski**
**Reference to related ticket:**
#40, #43 (backend part), #46
**Description of changes:**
 - `listMembers` event for listing members of a room
 - Documentation for `listMembers`

### CODE REVIEW

File: src/server/ws-routes/controllers/Room.js [hunk 1]
```diff
@@ -29,9 +30,9 @@ class Room {
 		const {roomId} = data
 		const username = socket.request.user.username
 		const timestamp = new Date().getTime()
-
-		socket.emit(EVENT_TYPES.JOIN, {username, roomId, timestamp})
+		socket.to(roomId).emit(EVENT_TYPES.JOIN, {username, roomId, timestamp})
 		socket.join(roomId)
```
Comment (michaelKurowski) [inline src/server/ws-routes/controllers/Room.js]: Swap `socket.to(roomId).emit(EVENT_TYPES.JOIN, {username, roomId, timestamp})` and `socket.join(roomId)`

**Comment by michaelKurowski** [comment]
TODO: Finish `strategies.spec.js`

File: src/server/.tests/passport/strategies.spec.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+const strategiesProvider = require('../../passport/strategies')
+const responseManager = require('../../routes/controllers/utilities/responseManager')
+const sinon = require('sinon')
+const assert = require('chai').assert
+
+let suite
+describe('passport strategies', () => {
+	beforeEach(() => {
+		suite = {}
+		suite.FAKE_FIND_BY_ID_RESULT = 'fakeUsr'
```
Comment (Jordan141) [inline src/server/.tests/passport/strategies.spec.js]: change to fakeUser

File: src/server/passport/strategyUtils.js [hunk 3]
```diff
@@ -32,15 +32,7 @@ function strategyHandlers(req, res, next) {
...
-	if (hash === user.password) {
-		const error = null
-		done(error, user)
-		return
-	}
-
-	const userToSerialize = null
-	done(responseManager.MESSAGES.errors.UNAUTHORIZED, userToSerialize)
-	return
+	return (hash === user.password) ? null : responseManager.MESSAGES.errors.UNAUTHORIZED
```
Comment (Jordan141) [inline src/server/passport/strategyUtils.js]: Change null to something meaningful

File: src/server/passport/strategyUtils.js [hunk 4]
```diff
@@ -32,7 +32,8 @@ function strategyHandlers(req, res, next) {
 
 function validateUserPassword(user, password, done) {
 	const hash = Utilities.createSaltedHash(user.salt, password)
-	return (hash === user.password) ? null : responseManager.MESSAGES.errors.UNAUTHORIZED
+	const error = (hash === user.password) ? null : responseManager.MESSAGES.errors.UNAUTHORIZED
```
Comment (Jordan141) [inline src/server/passport/strategyUtils.js]: I meant to add the true statement's null to something more readable

**Pull request merged**